lib/rucio/core/trace.py

```diff
@@ -335,26 +353,23 @@ def trace(payload):
         LOGGER.error(error)
 
 
-def validate_schema(obj):
+def validate_schema(obj: str) -> None:
     """
     Validate object against json schema
 
     :param obj: The object to validate.
 
-    :raises: InvalidObject
+    :raises: InvalidObject, TraceValidationSchemaNotFound
     """
-    obj = json.loads(obj)
+    loaded_obj: 'TraceSchemaDict' = json.loads(obj)
 
-    try:
-        if obj and 'eventType' in obj:
-            event_type = SCHEMAS.get(obj['eventType'].lower())
-            if not event_type:
-                validation_error = ValidationError(message=f"Trace schema for eventType {obj['eventType']} is not currently supported.")
-                validation_error.cause = "SCHEMA_NOT_FOUND"
-                raise validation_error
-            validate(obj, SCHEMAS.get(obj['eventType'].lower()), format_checker=FORMAT_CHECKER)
-    except ValidationError as error:
-        if error.cause == "SCHEMA_NOT_FOUND":
-            LOGGER.error(error)
+    if loaded_obj and 'eventType' in loaded_obj:
+        event_type = loaded_obj['eventType'].lower()
+        schema = SCHEMAS.get(event_type)
+        if schema is not None:
+            try:
+                validate(loaded_obj, schema, format_checker=FORMAT_CHECKER)
+            except ValidationError as error:
+                raise InvalidObject(error)
         else:
-            raise InvalidObject(error)
+            raise TraceValidationSchemaNotFound("Trace schema for eventType %s not found." % event_type)
```

Comment:
The added verbosity of "Trace scheme not supported" is better explanation of what the actual issue is I think. Sure, the schema couldn't be found, but it's because it's not supported, not because a schema file is missing somewhere.

Reply:
I agree, good point - I based the exception on the error cause that was previously present (`SCHEMA_NOT_FOUND`).

I think it depends on whether `eventType` has been validated before getting to `core` here. IMO the best setup would be this:
1. at the API level, `eventType` should be validated to see if it's a supported event type. If not, it raises `EventTypeNotSupported`
2. At the core level (here), assuming (due to point 1) that this can only be called with a valid `eventType`, we raise `TraceValidationSchemaNotFound` if we don't find a schema for this event type.

I'm not sure we're doing API validation at the moment, but I think ideally this might be preferred over raising `EventTypeNotSupported` here (and effectively doing validation in `core`)

Reply:
Yeah ideally this validation should be done in the API [but it doesn't seem to be](https://github.com/rucio/rucio/blob/master/lib/rucio/web/rest/flaskapi/v1/traces.py#L81). Unless you want to do that refactor in this PR it may be be just a nice band-aid to have the raised error more descriptive

Reply:
Expanded the error message in the latest revision so that it mentions that the error could either be that the trace schema was not found or that the event type is unsupported